File: text/csharp/update/text_update.cs
// ----------------------------------------------------------------
/*
	update/text_update.cs

					Jul/16/2016

*/
// ----------------------------------------------------------------
using System;
using System.Collections.Generic;

// ----------------------------------------------------------------
public static class text_update
{
// ----------------------------------------------------------------
public static int Main (string[] args)
{
	Console.Error.WriteLine ("*** 開始 ***");

	if (args.Length < 2)
		{
		Console.Error.WriteLine ("*** error ***");
		return 1;
		}

	string	file_txt = args[0];

	string	id_in = args[1];
	int	population_in = int.Parse (args[2]);

	Console.WriteLine (id_in + "\t" + population_in);

	Dictionary <string,Object> dict_aa
		= text_manipulate.text_read_proc (file_txt);

	dict_aa = text_manipulate.dict_update_proc
		(dict_aa,id_in,population_in);

	text_manipulate.text_write_proc (file_txt,dict_aa);

	Console.Error.WriteLine ("*** 終了 ***");

	return	0;
}

// ----------------------------------------------------------------
}
// ----------------------------------------------------------------

File: text/fsharp/update/text_update.fs
// ----------------------------------------------------------------
//	text_update.fs
//
//						Mar/09/2018
//
// ----------------------------------------------------------------
open System
open System.IO
open Microsoft.FSharp.Core

// ----------------------------------------------------------------
[<EntryPoint>]
let main (args : string[] ) =

  Console.Error.WriteLine("*** 開始 ***")

  let file_txt = args.[0]

  let id_in = args.[1]
  let population_in = int args.[2]
  printfn "%s\t%d" id_in population_in

  let dict_aa = text_manipulate.m01.text_read_proc (file_txt)

  let dict_aa = text_manipulate.m01.dict_update_proc (dict_aa,id_in,population_in)

  let _ = text_manipulate.m01.text_write_proc (file_txt,dict_aa)

  Console.Error.WriteLine("*** 終了 ***")
  0
// ----------------------------------------------------------------
